runtime/protocol/http-test/jvmAndNative/src/aws/smithy/kotlin/runtime/httptest/TestWithLocalServer.kt

```diff
@@ -19,7 +18,7 @@ import kotlin.time.Duration.Companion.seconds
  * mocking an HTTP client engine is difficult.
  */
 public abstract class TestWithLocalServer {
-    protected val serverPort: Int = ServerSocket(0).use { it.localPort }
+    protected val serverPort: Int = 54734
```

Comment:
**Question**: This will conflict if any other process (including a parallel build of this package in another workspace) is using port 54734. Is there a reason why it's necessary to use a static port?

Reply:
If we don't configure a port, [it will default to `80`](https://api.ktor.io/older/2.2.4/ktor-server/ktor-server-host-common/io.ktor.server.engine/embedded-server.html), we could make it a randomly selected port if you think that will be safer?

Reply:
Random ports are marginally safer but still could conflict with an open port. I think we can still bind to an available address using Ktor's low-level KMP socket interfaces:

```kotlin
protected val serverPort: Int = runBlocking {
    SelectorManager(coroutineContext).use { selector ->
        aSocket(selector)
            .tcp()
            .bind()
            .use { (it.localAddress as InetSocketAddress).port }
    }
}
```

Reply:
Nice, this works great